src/main/java/net/mcreator/ui/minecraft/states/item/JItemPropertiesListEntry.java

```diff
@@ -0,0 +1,131 @@
+import net.mcreator.element.parts.procedure.Procedure;
+import net.mcreator.ui.MCreator;
+import net.mcreator.ui.component.util.PanelUtils;
+import net.mcreator.ui.help.HelpUtils;
+import net.mcreator.ui.help.IHelpContext;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.init.UIRES;
+import net.mcreator.ui.minecraft.states.JPropertyNameField;
+import net.mcreator.ui.minecraft.states.PropertyData;
+import net.mcreator.ui.procedure.ProcedureSelector;
+import net.mcreator.ui.validation.IValidable;
+import net.mcreator.ui.validation.Validator;
+import net.mcreator.ui.validation.validators.ProcedureSelectorValidator;
+import net.mcreator.workspace.elements.VariableTypeLoader;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.List;
+
+public class JItemPropertiesListEntry extends JPanel implements IValidable {
+
+	private final JButton remove = new JButton(UIRES.get("16px.clear"));
+
+	private final JPropertyNameField nameField;
+
+	private final ProcedureSelector value;
+
+	public JItemPropertiesListEntry(MCreator mcreator, IHelpContext gui, JPanel parent,
+			List<JItemPropertiesListEntry> entryList, JPropertyNameField nameField, int propertyId) {
+		super(new BorderLayout(10, 5));
+		this.nameField = nameField;
+
+		setBorder(BorderFactory.createEmptyBorder(5, 5, 5, 5));
+
+		value = new ProcedureSelector(gui.withEntry("item/custom_property_value"), mcreator,
+				L10N.t("elementgui.item.custom_property.value"),
+				L10N.t("elementgui.item.custom_property.value") + propertyId, ProcedureSelector.Side.CLIENT, true,
+				VariableTypeLoader.BuiltInTypes.NUMBER,
+				Dependency.fromString("x:number/y:number/z:number/world:world/entity:entity/itemstack:itemstack"));
+		value.setValidator(new ProcedureSelectorValidator(value));
+		reloadDataLists(); // we make sure that selector can be properly shown
+
+		add("West", PanelUtils.pullElementUp(PanelUtils.northAndCenterElement(
+				HelpUtils.wrapWithHelpButton(gui.withEntry("item/custom_property_name"),
+						L10N.label("elementgui.item.custom_property.name")), nameField)));
+		add("Center", value);
+
+		JComponent container = PanelUtils.expandHorizontally(this);
+		parent.add(container);
+		entryList.add(this);
+
+		remove.setText(L10N.t("elementgui.item.custom_property.remove"));
+		remove.addActionListener(e -> {
+			parent.remove(container);
+			parent.revalidate();
+			parent.repaint();
+			entryList.remove(this);
+		});
+		add("East", PanelUtils.pullElementUp(remove));
+
+		parent.revalidate();
+		parent.repaint();
+	}
+
+	@Override public void setEnabled(boolean enabled) {
+		super.setEnabled(enabled);
+
+		nameField.setEnabled(enabled);
+		value.setEnabled(enabled);
+
+		remove.setEnabled(enabled);
+	}
+
+	public void reloadDataLists() {
+		value.refreshListKeepSelected();
+	}
+
+	JPropertyNameField getNameField() {
+		return nameField;
+	}
+
+	PropertyData.FloatNumber toPropertyData() {
+		return new PropertyData.FloatNumber(nameField.getText(), 0F, 1000000F);

```

Comment:
How was this range determined? Why can't one use 1000001 or maybe -1?

Reply:
Not sure why I chose this maximum, but for the negative values, I think guides were saying that they should not be used.

Reply:
Checking the source code it can be any float value, so Float.MIN_VALUE to Float.MAX_VALUE

Reply:
✔️ Fixed